### PR TITLE
gcp - doc for cloud router and delete action

### PR DESCRIPTION
**Pull request by PMitrafanau**

### CODE REVIEW

File: docs/source/gcp/examples/network-cloudrouter.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Network - Check if there are unattached Cloud Routers
+============================================================
```
Comment (kapilt) [inline docs/source/gcp/examples/network-cloudrouter.rst]: in general we should use the class documentation for reference examples, not every action/filter/resource needs an example in the folder, especially if its not a common usage scenario.

File: tools/c7n_gcp/c7n_gcp/resources/network.py [hunk 2]
```diff
@@ -140,9 +144,41 @@ def get(client, resource_info):
...
+            resource: gcp.router
+            filters:
+              - type: value
+                key: name
+                op: eq
+                value: test-router
+            actions:
+              - delete
+
+    https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroupManagers/delete
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/resources/network.py]: bad link, also please use sphinx/rst format for creating nice links.

File: docs/source/gcp/examples/network-cloudrouter.rst (hunk 1 above)
Comment (PMitrafanau) [reply docs/source/gcp/examples/network-cloudrouter.rst]: I would say it's rather use case for cost savings/keeping a project clean and well structured than a reference example.

Comment (kapilt) [reply docs/source/gcp/examples/network-cloudrouter.rst]: again in the real world this is pretty corner case not common case. how often have you seen projects with extraneous cloud routers ?

Comment (PMitrafanau) [reply docs/source/gcp/examples/network-cloudrouter.rst]: Fixed

File: tools/c7n_gcp/c7n_gcp/resources/network.py (hunk 2 above)
Comment (PMitrafanau) [reply tools/c7n_gcp/c7n_gcp/resources/network.py]: Fixed

**Pull request merged**